lib/loader/index.js

```diff
@@ -47,14 +47,14 @@ function getStringImpl(buffer, ptr) {
   let length = U32[(ptr + SIZE_OFFSET) >>> 2] >>> 1;
   let offset = ptr >>> 1;
   if (length <= CHUNKSIZE) return String.fromCharCode.apply(String, U16.subarray(offset, offset + length));
-  const parts = [];
+  let parts = '';
   do {
     const last = U16[offset + CHUNKSIZE - 1];
     const size = last >= 0xD800 && last < 0xDC00 ? CHUNKSIZE - 1 : CHUNKSIZE;
-    parts.push(String.fromCharCode.apply(String, U16.subarray(offset, offset += size)));
+    parts += String.fromCharCode.apply(String, U16.subarray(offset, offset += size));
```

Comment:
Are you sure that this is always faster? Would imagine that it is for smaller strings, but once `parts` grows bigger, it will might slow down substantially.

Reply:
On Chrome it should be definitely faster. See this tests:
https://twitter.com/MaxGraey/status/1232045301374111744
https://esbench.com/bench/5e540d53328a52009e8b1b22
It relate to string reversing but it also use same technique as string decoding

Reply:
You don't need to do `String.fromCharCode.apply(String)` could simply do something like

```ts
const fromCharCode = String.fromCharCode;

fromCharCode.apply(null, theArray);
```

Reply:
Does this affect performance?

Reply:
I don't think so

Reply:
@dcodeIO I made micro benchmark which more relate to this case:
https://gist.github.com/MaxGraey/6db5c6c6d7e4a0f45154b73a5e130976

Reply:
While 4K+ is already a reasonably long string, an interesting measurement there would be how long a string has to be for both approaches to execute in an equal time span, if at all. That would help to inform if it is good enough in any case, or if we'd need another code path to deal with strings longer than X.

Reply:
Also, the test string is exclusively ASCII currently, so it might be that engines optimize this differently than strings that have at least one non-LATIN1 character. Both cases would be interesting.

Reply:
Based on my benchmarks inside Node, `concat` is faster than `+`:
https://github.com/aminya/typescript-optimization#string--vs-concat